gateway/api/guardrails/guardrails.go

```diff
@@ -125,16 +191,21 @@ func List(c *gin.Context) {
 		c.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
 		return
 	}
+
 	rules := []openapi.GuardRailRuleResponse{}
 	for _, rule := range ruleList {
+		// Get connections for each rule
+		connectionIDs, _ := models.GetConnectionIDsForGuardRail(ctx.GetOrgID(), rule.ID)
```

Comment:
This is very bad because you're opening a connection for each record. I'm going to let this pass for now, but make sure to perform what you want in SQL and just mapping the types properly.

What I recommend is refactoring the logic of Updating, Creating and Listing to include the connection ids instead of writing a separate function to fetch parts of this resources.